assets/src/stories-editor/components/contexts/snapping.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { SVG } from '@wordpress/components';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { createContext, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { STORY_PAGE_INNER_HEIGHT, STORY_PAGE_INNER_WIDTH } from '../../constants';
+
+const SnapContext = createContext();
+
+const Snapping = ( { children } ) => {
+	const [ snapLines, setSnapLines ] = useState( [] );
+	const [ hasSnapLines, setHasSnapLines ] = useState( false );
```

Comment:
Thinking a bit more about it, is this variable even necessary? Can't we stick to the array only and if empty, show nothing? We always `hide` and `clear` at the same time anyway I think. Is there any occasion where we don't show the lines but remember them anyway?

Reply:
Hmm I can't think of any such situation at the moment.

One thing that comes to mind: When we implement the actual snapping, we _could_ still show the snapping guidelines, but disable the actual snapping when pressing <kbd>Alt</kbd>. But that does not affect this case here.

So I think it's fine to remove.

Reply:
Removed it now

Reply:
Perfect, that makes the API a lot cleaner.